shard/committee/assignment.go

```diff
@@ -129,6 +129,20 @@ func prepareOrders(
 	essentials := map[common.Address]*effective.SlotOrder{}
 	totalStaked, tempZero := big.NewInt(0), numeric.ZeroDec()
 
+	// Avoid duplicate BLS keys as harmony nodes
+	instance := shard.Schedule.InstanceForEpoch(stakedReader.CurrentBlock().Epoch())
```

Comment:
why would there be duplicates?

Reply:
In case our key is mistakenly used as validators . It may cause nil pointer crash